newshour-transcript-sync/process.py

```diff
@@ -18,37 +18,97 @@
+            mappings.update({int(i1): {'srt': " ".join(tokens1[i1:i2]), 'txt': " ".join(tokens2[i1:i2])}})
+        else:
+            mappings.update({int(i1): {'srt': " ".join(tokens1[i1:i2]), 'txt': " ".join(tokens2[j1:j2])}})
 
+    return mappings
 
-def srt_to_tsv(srt_filename, tsv_filename):
+
+def srt_to_tsv(srt_filename, gold_transcript_filename, tsv_filename):
+    """
+    Given a srt file and its corresponding gold transcript file,
+    replace the srt text with the gold transcript corresponding texts
+    and output it to the given tsv file.
+    """
     subs = pysrt.open(srt_filename, encoding='utf-8')
+    non_gold = ""
+    for sub in subs:
+        non_gold += sub.text.replace('\n', ' ')
+    with open(gold_transcript_filename, 'r') as f:
+        gold = f.read()
+
+    mappings = get_tokens_aligned(non_gold, gold)
+    prev_length = 0
+    current_length = 0
+
     with open(tsv_filename, 'w', encoding='utf-8') as out:
         out.write("index\tstart\tend\tcontent\n")
         # sub means subtitles
         for sub in subs:
             index = sub.index
             start = str(sub.start).replace(',', '.')
             end = str(sub.end).replace(',', '.')
-            text_content = sub.text.replace('\n', ' ')
+            replaced_line = ""
+            current_length += len(sub.text.replace('\n', ' ').split())
+            for i in range(prev_length, current_length):
+                try:
+                    replaced_line += mappings[i]['txt'] + " "
+                except KeyError:
+                    replaced_line += " "
+            text_content = replaced_line
+            prev_length = current_length
+
             out.write(f"{index}\t{start}\t{end}\t{text_content}\n")
 
 
-def convert_directory(source_directory, destination_directory):
+def convert_directory(source_directory, destination_directory, gold_transcript_directory):
     if not os.path.exists(destination_directory):
         os.makedirs(destination_directory)
     for filename in os.listdir(source_directory):
         if filename.endswith('.srt'):
-            source_path = os.path.join(source_directory, filename)
-            dest_path = os.path.join(destination_directory, filename.replace('.srt', '.tsv'))
-            srt_to_tsv(source_path, dest_path)
+            gold_transcript_path = filename.split(".")[0]+".txt"
+            if gold_transcript_path in os.listdir(gold_transcript_directory):
+                source_path = os.path.join(source_directory, filename)
+                dest_path = os.path.join(destination_directory, filename.replace('.srt', '.tsv'))
+                srt_to_tsv(source_path, dest_path, gold_transcript_path)
+            else:
+                print(filename, "has no corresponding gold transcript in", gold_transcript_directory)
 
 
 if __name__ == '__main__':
+    arg_parser = argparse.ArgumentParser()
+    arg_parser.add_argument('--goldTranscriptsDir', required=True,
```

Comment:
let's do something like `--gold-txt-dir` the name style (`goldTranscriptsDir`) isn't very pythonic.